File: JanperCab/5-JanperCab/ClientApp/src/app/_models/hinge-hole-option/HingeHoleOptionDto.ts
import { HingeHoleStyleEnum } from '../../_enums/HingeHoleStyleEnum';
import { FormGroup } from '@angular/forms';
import { DuraformComponentWithOptionAndHingeHoleDto } from '../duraform-component/DuraformComponentWithOptionAndHingeHoleDto';

export abstract class HingeHoleOptionDto {
  $type: string;
  id: number;
  hingeHoleStyle: HingeHoleStyleEnum;

  constructor() {
    this.$type = '';
  }

  abstract getPrice(
    component: DuraformComponentWithOptionAndHingeHoleDto
  ): number;
  abstract update(hingeHoleFormValue: any): void;
  abstract toFormGroup(): FormGroup;
  abstract toString(): string;
}

File: JanperCab/3-Application/Dtos/HingeHoleOption/HingeHoleOptionDto.cs
using _1_Domain.Enum;

namespace _3_Application.Dtos.HingeHoleOption
{
    public abstract class HingeHoleOptionDto
    {
        public int Id { get; set; }

        public HingeHoleStyleEnum HingeHoleStyle { get; set; }
    }
}